### ISSUE TITLE
[cloud_firestore] Incorrect DocumentReference == equals behavior on the same path on default firestore instance

### ISSUE THREAD

**Issue by duttaoindril** [FirebaseExtended/flutterfire#2081]
**Describe the bug**
When doing an equals on two document references:

```
DocumentReference docA = Firestore.instance.collection(“foo”).document(“bar”);
DocumentReference docB = Firestore.instance.collection(“foo”).document(“bar”);

print(docA == docB); // false
```

When diving into the issue a bit, I found the [source code](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore/lib/src/document_reference.dart#L24) for DocumentReference `==` checks the path and the firestore instance.

The path is equal, so when I looked at the two firestore instances, even though they're both from the same default app, (only one firebase app in my project), they're not equal.

```
// ...continued from above

print(docA.firestore == docB.firestore); // false
```

I didn't get a chance to dive deeper into why the above equals check doesn't work, but upon inspecting the firestore instances, I saw the default firebase app doesn't have the same hashcode across the two firestore instances 🤔

![image](https://user-images.githubusercontent.com/4969854/75499430-cdb27f00-597e-11ea-8d6e-74542957eed2.png)

That might be a helpful tip. Maybe look into the equals for firestore / firebase? Maybe always kick back true if both firestore instances are from the default app?

**To Reproduce**
1. Run the code above and see the output

**Expected behavior**
Document references with the same path from the same default firestore instance should be equal.

**Additional context**
This may or may not be also messing up usage of `DocumentReference`s as a viable key in a map - lookups to maps with the `DocumentReference` as it's key randomly doesn't work, which may or may not be due to this issue.

**Comment by Ehesp** [comment]
The equal check on the instance should be checking the `name` property, since the reference of the two instances is different (hence the == failing).

**Comment by duttaoindril** [comment]
Should I make a PR?

Thank you!
Sincere Regards,
Oindril Dutta <http://goog_799414691>
odutta.me

**Comment by Salakar** [comment]
Hey @duttaoindril, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you for raising this issue
